pkg/backends/calico/routes_test.go

```diff
@@ -0,0 +1,132 @@
+package calico
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func addEndpointSubset(ep *v1.Endpoints, nodename string) {
+	ep.Subsets = append(ep.Subsets, v1.EndpointSubset{
+		Addresses: []v1.EndpointAddress{
+			v1.EndpointAddress{
+				NodeName: &nodename}}})
+}
+
+func buildSimpleService() (rg *routeGenerator, svc *v1.Service, ep *v1.Endpoints) {
+	rg = &routeGenerator{
```

Comment:
I'd remove this from here - this _does_ belong in BeforeEach(), since this is the unit under test :)